### DISCUSSION TITLE
php 8.0

### DISCUSSION THREAD

**Discussion by diderich** [YetiForceCompany/YetiForceCRM#14890] [How to]
At some time in the past I remember to be reading that version 6.2. will work with php 8.0. Is this the case, or do we still need php 7.4 to run the system successfully. At that time it seemed that not all libraries used were 8.0-ready.

**Comment by mariuszkrzaczkowski** [comment]
Libraries should be PHP 8 compatible, due to the fact that we have ditched PHP 7.2.
We have not checked if the system works on PHP 8, we will be in this version.

Unit testing on PHP 8 works fine, you would have to give it a try :)

**Reply by mariuszkrzaczkowski** [reply]
https://github.com/YetiForceCompany/YetiForceCRM/runs/2971136368?check_suite_focus=true

**Comment by diderich** [comment]
I have encountered the following warning when using php 8.0.8
```

: uksort(): Returning bool from comparison function is deprecated, return an integer less than, equal to, or greater than zero in /opt/crm6/app/QueryGenerator.php, line 971
#0 app/QueryGenerator.php (971)  >>  uksort([array,array])
#1 app/QueryGenerator.php (824)  >>  App\QueryGenerator->loadJoin()
#2 modules/Vtiger/models/RelationListView.php (195)  >>  App\QueryGenerator->createQuery()
#3 modules/Vtiger/models/RelationListView.php (251)  >>  Vtiger_RelationListView_Model->getRelationQuery()
#4 modules/Vtiger/views/Detail.php (894)  >>  Vtiger_RelationListView_Model->getEntries()
#5 app/Controller/ExposeMethod.php (68)  >>  Vtiger_Detail_View->showRelatedRecords()
#6 modules/Vtiger/views/Detail.php (181)  >>  Vtiger_Detail_View->invokeExposedMethod('showRelatedRecords')
#7 include/main/WebUI.php (190)  >>  Vtiger_Detail_View->process()
#8 index.php (26)  >>  Vtiger_WebUI->process()
#9 public_html/index.php (11)  >>  require('index.php')

```

**Reply by mariuszkrzaczkowski** [reply]
https://github.com/YetiForceCompany/YetiForceCRM/issues/14905

**Reply by mariuszkrzaczkowski** [reply]
We switched CRM https://gitdeveloper.yetiforce.com/ to php 8.0.9, you can test

**Comment by mariuszkrzaczkowski** [comment]
roundcube mail does not support php 8 yet
`PHP version 5.4.1 or greater (and < 8.0 **) with`
https://github.com/roundcube/roundcubemail/wiki/Install-Requirements#server-requirements
